vizro-core/src/vizro/models/_page.py

```diff
@@ -151,10 +151,8 @@ def pre_build(self):
         if targets:
             self.actions = [_on_page_load(id=f"{ON_PAGE_LOAD_ACTION_PREFIX}_{self.id}", targets=targets)]
 
-        controls = cast(
-            Iterable[ControlType],
-            [*model_manager._get_models(Parameter, self), *model_manager._get_models(Filter, self)],
-        )
+        # Convert generator to list as it's going to be iterated multiple times.
```

Comment:
This is the bit (and comparable code in container) where I think ideally we'd look at child models of `self.controls` rather than `self` (which includes `self.anything`). I know the `_get_models` stuff is probably not currently well setup to do that though, so if it's way too complicated then don't worry about it. Something like this maybe? With some comment explaining what's going on.
```
controls = []
for control in self.controls:
    if isinstance(control, (Filter, Parameter)):
        controls.append(control)
    else:
        controls.extend(model_manager._get_models((Filter, Parameter), control))
```

Reply:
Ahh yes, that makes perfect sense, thank you! Please could you add a comment here about why we need to look in `self` and not just `self.controls`.